### ISSUE TITLE
vi mode - normal and visualindicated the same way

### ISSUE THREAD

**Issue by raffaem** [randy3k/radian#462]
With
```
options(radian.editing_mode = "vi")
options(radian.show_vi_mode_prompt = TRUE)
options(radian.vi_mode_prompt = "\033[0;34m[{}]\033[0m ")
```

while insert mode is indicated with "ins", both normal and visual mode are indicated in the same way, with "nav".

On a side note:
1. When in visual mode, pressing ESC takes a lot of time to exit visual mode (1/2 seconds)
2. Is it possible to indicate the mode with just one letter (i/n/v)?

**Comment by randy3k** [comment]
> both normal and visual mode are indicated in the same way, with "nav".
It is expected behavior as long as i can tell. 

> When in visual mode, pressing ESC takes a lot of time to exit visual mode (1/2 seconds)
it is a known issue of prompt-toolkit.  `ipython` shares the same issues.

>  Is it possible to indicate the mode with just one letter (i/n/v)?
```
options(
    radian.vi_mode_prompt = list(
        "ins" = "(I) ",
        "nav" = "(N) ",
        "rep" = "(R) "
    )
)
```

**Comment by raffaem** [comment]
> > Is it possible to indicate the mode with just one letter (i/n/v)?
> 
> ```
> options(
>     radian.vi_mode_prompt = list(
>         "ins" = "(I) ",
>         "nav" = "(N) ",
>         "rep" = "(R) "
>     )
> )
> ```

Is this documented?

**Comment by raffaem** [comment]
uhm can I have per-mode cursors? I would like a box in normal mode, and a vertical bar in insert mode.

As you can see, I am very, very spoiled 😛

**Comment by raffaem** [comment]
> > When in visual mode, pressing ESC takes a lot of time to exit visual mode (1/2 seconds)
> 
> it is a known issue of prompt-toolkit. `ipython` shares the same issues.

is it ever going to be fixed?

I wrote a bug report and never received an answer

**Comment by randy3k** [comment]
> Is this documented?

No. But it is not likely to be changed unless prompt-toolkit changes the input modes.


> uhm can I have per-mode cursors? I would like a box in normal mode, and a vertical bar in insert mode.

prompt-toolkit added this support in https://github.com/prompt-toolkit/python-prompt-toolkit/pull/1558

it should be doable, but I don't have the time to invest in it now.

**Comment by randy3k** [comment]
> is it ever going to be fixed?
> 
> I wrote a bug report and never received an answer

I don't think it is fixable without major refactoring in prompt-toolkit's key capturing mechanism.
